fire_atlas_editor/lib/vendor/micro_store/micro_store.dart

```diff
@@ -50,29 +50,29 @@ typedef MicroStoreProviderMemo<T> = List Function(MicroStore<T>);
 class MicroStoreProvider<T> extends StatefulWidget {
   final MicroStore<T> store;
   final MicroStoreProviderBuilder<T> builder;
-  final MicroStoreProviderMemo<T> memoFn;
+  final MicroStoreProviderMemo<T>? memoFn;
 
   MicroStoreProvider({
-    Key key,
-    this.store,
-    this.builder,
+    required this.store,
+    required this.builder,
     this.memoFn,
+    Key? key,
   }) : super(key: key);
 
   @override
   State createState() => _MicroStoreProviderState<T>();
 }
 
 class _MicroStoreProviderState<T> extends State<MicroStoreProvider<T>> {
-  MicroStore<T> _store;
-  List _memoValue;
+  late MicroStore<T> _store;
+  List? _memoValue;
 
   void _update(MicroStore<T> store) {
     Function eq = const ListEquality().equals;
 
-    List newMemo;
+    List? newMemo;
     if (widget.memoFn != null) {
-      final newMemo = widget.memoFn(store);
+      final newMemo = widget.memoFn!(store);
```

Comment:
I think this is a bug, did you mean to use the outer variable? this just declares a new one, the outer one is always null

Reply:
Good catch